R/extract_data.R

```diff
@@ -0,0 +1,245 @@
+  database$contexts <- database$contexts %>% tidyr::separate_longer_delim(link_vals, delim = ", ")
+
+  database$contexts_tmp <- split(database$contexts, database$contexts$link_id)
+  
+  empty_tibble <- dplyr::tibble(
+    dataset_id = NA_character_,
+    context_property = NA_character_,
+    category = NA_character_,
+    value = NA_character_,
+    description = NA_character_,
+    link_id = NA_character_,
+    link_vals  = NA_character_
+  )
+
+  if (is.null(database$contexts_tmp$entity_context_id)) {
+    database$contexts_tmp$entity_context_id <- empty_tibble
+  }
+  
+  if (is.null(database$contexts_tmp$method_context_id)) {
+    database$contexts_tmp$method_context_id <- empty_tibble
+  }
+  
+  if (is.null(database$contexts_tmp$temporal_context_id)) {
+    database$contexts_tmp$temporal_context_id <- empty_tibble
+  }
+  
+  if (is.null(database$contexts_tmp$plot_context_id)) {  
+    database$contexts_tmp$plot_context_id <- empty_tibble
+  }
+  
+  if (is.null(database$contexts_tmp$treatment_context_id)) {
+    database$contexts_tmp$treatment_context_id <- empty_tibble
+  }
+  
+  database$entity_context_id <- database$contexts_tmp$entity_context_id %>%
+    dplyr::rename(entity_context_id = link_vals)
+  
+  database$method_context_id <- database$contexts_tmp$method_context_id %>%
+    dplyr::rename(method_context_id = link_vals)  
+  
+  database$temporal_context_id <- database$contexts_tmp$temporal_context_id %>%
+    dplyr::rename(temporal_context_id = link_vals)
+  
+  database$plot_context_id <- database$contexts_tmp$plot_context_id %>%
+    dplyr::rename(plot_context_id = link_vals)
+  
+  database$treatment_context_id <- database$contexts_tmp$treatment_context_id %>%
+    dplyr::rename(treatment_context_id = link_vals)
+
+  
+  tables_to_cut <- c("locations", "entity_context_id", "method_context_id", "temporal_context_id", 
+                            "plot_context_id", "treatment_context_id", 
+                            "methods", "taxa", "taxonomic_updates", "contributors")
+  
+  tables_complete_path <- c("database$locations", "database$entity_context_id", 
+                            "database$method_context_id", "database$temporal_context_id", 
+                            "database$plot_context_id", "database$treatment_context_id", 
+                            "database$methods", "database$taxa", "database$taxonomic_updates", "database$contributors")
+  
+  # Empty list
+  ret <- list(
+    traits = dplyr::tibble(), 
+    locations = dplyr::tibble(),
+    contexts = dplyr::tibble(),
+    entity_context_id = dplyr::tibble(),
+    method_context_id = dplyr::tibble(),
+    temporal_context_id = dplyr::tibble(),
+    plot_context_id = dplyr::tibble(),
+    treatment_context_id = dplyr::tibble(),
+    methods = dplyr::tibble(),
+    excluded_data = dplyr::tibble(),
+    taxonomic_updates = dplyr::tibble(),
+    taxa = dplyr::tibble(),
+    contributors = dplyr::tibble()
+  )
+  
+  ret_tmp <- list()
+
+  # Cookie cutters
```

Comment:
A named list would be ideal here:

```
# Cookie cutters
  cookie_cutters <- list(
  locations_cc = c("dataset_id", "location_id"),
  entity_contexts_cc = c("dataset_id", "entity_context_id"),
  temporal_contexts_cc = c("dataset_id", "temporal_context_id"),
  method_contexts_cc = c("dataset_id", "method_context_id"),
  plot_contexts_cc = c("dataset_id", "plot_context_id"),
  treatment_contexts_cc = c("dataset_id", "treatment_context_id"),
  methods_cc = c("dataset_id", "trait_name", "method_id"),
  taxonomic_updates_cc = c("dataset_id", "taxon_name", "original_name"),
  taxa_cc = c("taxon_name"),
  contributors_cc = c("dataset_id")
)



 ```

Reply:
I've got that going and there were a lot of flow-on problems, but by solving them I also got rid of the back-ticks that were creating other problems